Feat/components/modals and loading screen

gooddinosaur opened the pull request:
## Checklist
- [x] Add all modals
- [x] Loading screen
- [x] Navbar
- [x] Footer

## Screenshots / Demos
- Success Modal
<img width="444" height="434" alt="image" src="https://github.com/user-attachments/assets/638b9c18-ed01-459a-8bdc-7a779818f23b" />

- Unsuccess Modal
<img width="446" height="431" alt="image" src="https://github.com/user-attachments/assets/f7fa53a9-3afe-4cc4-9d87-e8dd2b649dda" />

- Confirm Modal
<img width="444" height="452" alt="image" src="https://github.com/user-attachments/assets/ef45234e-5d22-4490-8b02-db2df8ef96a6" />

- Delete Modal
<img width="442" height="454" alt="image" src="https://github.com/user-attachments/assets/9d9e58f6-8b11-413e-af01-93379ab89173" />

- Approve Modal
<img width="444" height="460" alt="image" src="https://github.com/user-attachments/assets/c99647ea-0a36-46ef-987a-f96f0d17b19d" />

- Warning Modal
<img width="442" height="480" alt="image" src="https://github.com/user-attachments/assets/7c014f66-22d3-4e4e-bbc8-b559c2524863" />

- Report Modal
<img width="448" height="340" alt="image" src="https://github.com/user-attachments/assets/fa338f0d-438f-4270-abc5-363d4ce413c8" />

- Loading Screen

https://github.com/user-attachments/assets/873e7a8b-e674-46ff-867f-1e9131753822

- Navbar
<img width="1899" height="74" alt="image" src="https://github.com/user-attachments/assets/b2a3ee16-60ed-466a-bdc7-3398aa71b0e1" />

- Footer
<img width="1894" height="71" alt="image" src="https://github.com/user-attachments/assets/d057e397-5881-4a9c-baf7-fa051ab90989" />

Fixed: #27, #28, #31, #39

gooddinosaur committed:
feat(register): add first version of company register page

gooddinosaur committed:
feat(component): add navbar and footer

gooddinosaur committed:
chore(component): add shadcn components

gooddinosaur committed:
chore(color): add color in globals.css

gooddinosaur committed:
feat(register): make upload file working

gooddinosaur committed:
feat(modal): Add all modals

gooddinosaur committed:
fix(modals): add readonly in interface of all modals

gooddinosaur committed:
feat(ui): add loading page

gooddinosaur committed:
chore(name): rename footer and navbar to follow pascal case

gooddinosaur committed:
chore(name): rename footer.tsx to Footer.tsx

gooddinosaur committed:
chore(name): rename navbar.tsx to NavBar.tsx

gooddinosaur committed:
chore(file): delete src/app/company-register directory

catwithasword reviewed (changes requested):
1. There's import alias in out project so you should properly use the import like @/components/ui/button. It is more appropriate.
2. I don't think you should push the modal-demo page tbh. it is not in the product.
3. My suggestion in coding is you should implement an index.ts file in /component/modals since there are many modal files and we may need to use it quite a lot and not just a single modal in one page, it is better if we can import it simply. (apply this with /ui as well).
4. Your checklist has mentioned Navbar and Footer but there is no screenshots of these components.

gooddinosaur committed:
chore(file): delete src/app/modal-demo directory

gooddinosaur committed:
fix(import): use use the import like @/components/ui/button

gooddinosaur committed:
fix(file): add index.ts in modals dir and ui dir

gooddinosaur committed:
Merge branch 'feat/components/modals-and-loading-screen' of https://github.com/HireMeMaybe/HireMeMaybe-frontend into feat/components/modals-and-loading-screen

gooddinosaur commented:
- I've updated the import paths to use the alias (e.g., @/components/ui/button) as recommended.
- Removed the modal-demo page since it's not part of the product.
- Added an index.ts file in both /components/modals and /components/ui for cleaner and more scalable imports.
- Also included screenshots of the Navbar and Footer as mentioned in the checklist.

catwithasword commented:
can you change the layout in warning modal? The save button should be on the right side since it has a better user experience (people usually click on the right button). People might lose their changes if it is on the left side. And it would be better if you changed the yellow color to be more orange. It is hard to see (and flashy) with the current color.

gooddinosaur committed:
fix(warning): change layout of warning modal and change yellow color to be more orange

gooddinosaur commented:
I’ve completed the task, updated the button layout and adjusted the color to a more orange tone.

- Warning Modal
<img width="619" height="679" alt="image" src="https://github.com/user-attachments/assets/81495197-a5ba-4935-b61c-dfc9c6a853ff" />

catwithasword reviewed (approved):
Everything looks good
